File: 4.정렬/BOJ/(10825번)국영수/[jeunghun] solution.swift
let n = Int(readLine()!)!
var scores = [(kor: Int, eng: Int, math: Int, name: String)]()

for _ in 0..<n {
    let data = readLine()!.split(separator: " ")
    scores.append((Int(data[1])!, Int(data[2])!, Int(data[3])!, String(data[0])))
}

let sortedScores = scores.sorted {
    if $0.kor != $1.kor {
        return $0.kor > $1.kor
    } else {
        if $0.eng != $1.eng {
            return $0.eng < $1.eng
        } else {
            if $0.math != $1.math {
                return $0.math > $1.math
            } else {
                return $0.name < $1.name
            }
        }
    }
}

for i in 0..<n {
    print(sortedScores[i].3)
}

File: 4.정렬/BOJ/(10825번)국영수/[jeunghun] solution.py
n = int(input())
scores = []

for _ in range(n):
    name, kor, eng, math = input().split()
    scores.append((int(kor), int(eng), int(math), name))

scores.sort(key=lambda x: (-x[0], x[1], -x[2], x[3]))

for i in range(n):
    print(scores[i][3])